packages/express/src/build.ts

```diff
@@ -13,25 +16,59 @@ export const build: BuildV3 = async args => {
     ...args,
     entrypoint,
     considerBuildCommand: true,
-    entrypointCallback: (preparedFiles: Files) => {
+    entrypointCallback: (preparedFiles: Files | Record<string, FileFsRef>) => {
       return findEntrypoint(preparedFiles);
     },
   });
 };
 
-export const findEntrypoint = (files: Files) => {
-  const validFilenames = [['app'], ['index'], ['server'], ['src', 'index']];
+export const findEntrypoint = (files: Files | Record<string, FileFsRef>) => {
+  const validFilenames = [
+    ['app'],
+    ['index'],
+    ['server'],
+    ['src', 'app'],
+    ['src', 'index'],
+    ['src', 'server'],
+  ];
 
-  const validExtensions = ['js', 'cjs', 'mjs', 'ts', 'mts'];
+  const validExtensions = ['js', 'cjs', 'mjs', 'ts', 'mts', 'cts'];
```

Comment:
Just to keep the same ordering as the `js` variants
```suggestion
  const validExtensions = ['js', 'cjs', 'mjs', 'ts', 'cts', 'mts'];
```